futaba/cogs/welcome/prune.py

```diff
@@ -0,0 +1,120 @@
+#
+# cogs/welcome/prune.py
+#
+# futaba - A Discord Mod bot for the Programming server
+# Copyright (c) 2017-2018 Jake Richardson, Ammon Smith, jackylam5
+#
+# futaba is available free of charge under the terms of the MIT
+# License. You are free to redistribute and/or modify it under those
+# terms. It is distributed in the hopes that it will be useful, but
+# WITHOUT ANY WARRANTY. See the LICENSE file for more details.
+#
+
+"""
+Functions to prune a user either manually or automatically
+"""
+
+import logging
+from datetime import datetime, timedelta
+
+import discord
+from discord.ext import commands
+
+from futaba import permissions
+
+logger = logging.getLogger(__name__)
+
+__all__ = ["Prune"]
+
+
+def prune_filter(member, prune_date, role, has_role=True):
+    """
+    Fuction used to filter members by.
+    If has_role is False it will check if the user dosen't have the role specified
+    """
+    if has_role:
+        if role in member.roles:
+            if member.joined_at < prune_date:
+                return True
+    elif role not in member.roles:
+        if member.joined_at < prune_date:
+            return True
```

Comment:
You could rewrite this to eliminate the second `if joined ...`.

Get `has_role = role in member.roles` (rename the argument to like `should_have_role` or something)
Then say `if has_role == should_have_role and member.joined_at < prune_date:`